packages/core/src/parser/util.ts

```diff
@@ -506,6 +507,77 @@ export function stopBefore<N extends Returnable>(
 	}
 }
 
+/**
+ * @returns A parser that is based on the passed-in `parser`, but concatenates lines
+ * together when we reach, in order:
+ * - a backslash
+ * - whitespace (optional)
+ * - a newline
+ * - whitespace (optional)
+ */
+export function concatOnTrailingBackslash<N extends Returnable>(
+	parser: InfallibleParser<N>,
+): InfallibleParser<N>
+export function concatOnTrailingBackslash<N extends Returnable>(
+	parser: Parser<N>,
+): Parser<N>
+export function concatOnTrailingBackslash<N extends Returnable>(
+	parser: Parser<N>,
+): Parser<N> {
+	return (src, ctx): Result<N> => {
+		let wrappedStr = src.sliceToCursor(0)
+		const wrappedSrcCursor = wrappedStr.length
+		const indexMap: IndexMap = []
+
+		while (src.canRead()) {
+			wrappedStr += src.readUntil('\\')
+			if (!src.canRead()) {
+				break
+			}
+
+			// If we get here, then `src.cursor` is at a backslash
+			if (src.hasNonSpaceAheadInLine(1)) {
+				wrappedStr += src.read()
+				continue
+			}
+
+			// Create an index map that skips from the trailing backslash to the
+			// next line's first non-whitespace character
+			const from = src.getCharRange()
+			src.nextLine()
+
+			// Minecraft raises a `Line continuation at end of file` if a backslash
+			// (+ optional whitespace to the next line) is right before the end of the file
+			if (!src.canRead()) {
+				const ans: ErrorNode = {
+					type: 'error',
+					range: Range.span(from, src),
+				}
+				ctx.err.report(
+					localize('mcfunction.parser.line-continuation-end-of-file'),
```

Comment:
Either move this parser to the `mcfunction` package, make this a more generic core parser message, or allow the function caller to supply a message.

Reply:
[`163283e` (#1180)](https://github.com/SpyglassMC/Spyglass/pull/1180/commits/163283ec04e09f7317fb45c494a1b6672843df57)